### PR TITLE
Issue 14 create a generic button component

### LINKED ISSUE #14: Create a Generic Button Component
## Basic Information
Create a reusable button component. This component should be customizable and accept two main props: the button text and a callback function to handle button clicks. Additionally, the button should have a default border-radius of 8px and support styling through Tailwind CSS for flexibility (e.g., solid or outlined styles).
[button design](https://www.figma.com/design/bhSe5HkMJKKN0Bs1ZqSddi/Penni?node-id=3471-5828&t=w0t63mb6QaHbIOjN-0)
## Requirements:

1. Props:
     - label: The label text to display on the button.
     - onClick: The callback function to execute when the button is clicked.
2. Default Styling:
    - [Border-radius](https://www.figma.com/design/bhSe5HkMJKKN0Bs1ZqSddi/Penni?node-id=449-241&t=w0t63mb6QaHbIOjN-0): 8px.

### PR DESCRIPTION
**Pull request by SeiKasahara**
## Change Summary
**Created a generic button component with the requirement of penni design**
**Added CSS attributes**

### Change Form

- [x] The pull request title has an issue number
- [ ] The change works by "Smoke testing" or quick testing
- [x] The change has tests
- [ ] The change has documentation

# Related issue

- Resolve #14

### CODE REVIEW

File: client/src/components/ui/button.tsx [hunk 1]
```diff
@@ -37,17 +41,24 @@ export interface ButtonProps
   extends React.ButtonHTMLAttributes<HTMLButtonElement>,
     VariantProps<typeof buttonVariants> {
   asChild?: boolean;
+  label?: string; // Added label attribute, should be type, and it's optional
+  children?: React.ReactNode; // If there is no label, use children
 }
-
+// Use label attribute
```
Comment (GlendleNguyen) [inline client/src/components/ui/button.tsx]:
```suggestion
```

File: client/src/components/ui/button.tsx [hunk 2]
```diff
@@ -37,17 +41,24 @@ export interface ButtonProps
   extends React.ButtonHTMLAttributes<HTMLButtonElement>,
     VariantProps<typeof buttonVariants> {
   asChild?: boolean;
+  label?: string; // Added label attribute, should be type, and it's optional
```
Comment (GlendleNguyen) [inline client/src/components/ui/button.tsx]:
```suggestion
  label?: string; // Optional label for the button
```

File: client/.env.example
Comment (GlendleNguyen) [inline client/.env.example:1]: Should we delete this?

Comment (SeiKasahara) [reply client/.env.example:1]: I think it's reasonable to let everybody know more clearly what's going on in this code.

File: client/tailwind.config.ts [hunk 3]
```diff
@@ -23,6 +23,7 @@ const config = {
         sans: ["var(--font-sans)", ...fontFamily.sans],
       },
       colors: {
+        customPrimary: "hsl(var(--customPrimary))",
```
Comment (yunho7687) [inline client/tailwind.config.ts]: you could use `penni-main:"#0051CA"` here to make it more semantic. 
good job tho

Comment (SeiKasahara) [reply client/tailwind.config.ts]: I found variables like "`--customPrimary`" should be included in `global.css`, I added this var in `global.css` like the 16th line: `--customPrimary: 225, 100%, 40%; /*#0051CA Penni Blue*/` 👀

File: client/.env.example
Comment (GlendleNguyen) [reply client/.env.example:1]: This was for the .env.example file

Comment (SeiKasahara) [reply client/.env.example:1]: For this I have no idea

File: client/tailwind.config.ts (hunk 3 above)
Comment (yunho7687) [reply client/tailwind.config.ts]: add `"penni-main“:"#0051CA"` here and use it like `className="bg-penni-main"`
we use tailwind css so, from my perspective, we should configure the theme color in `tailwind.config.ts`

File: client/.env.example
Comment (GlendleNguyen) [reply client/.env.example:1]: Ah, I'd recommend leaving this in so that other devs can use this as a template for setting up their projects 😄

Comment (SeiKasahara) [reply client/.env.example:1]: No worries I'll add it back

File: client/tailwind.config.ts (hunk 3 above)
Comment (SeiKasahara) [reply client/tailwind.config.ts]: I'll modify it

**Review by yunho7687 — COMMENTED** [review]
please use dash-case for the CSS

**Review by yunho7687 — APPROVED** [review]

**Pull request merged**